doc/source/user/absolute_beginners.rst

```diff
@@ -0,0 +1,2451 @@
+
+The four values listed above correspond to the number of columns in your array. With a four-column array, you will get four values as your result.
+
+`Read more about functions here <https://docs.scipy.org/doc/numpy/reference/arrays.ndarray.html>`_ and `calculations here <https://docs.scipy.org/doc/numpy-1.17.0/reference/arrays.ndarray.html#calculation>`_.
+
+
+How to inspect the size and shape of a NumPy array
+--------------------------------------------------
+
+
+::
+
+
+  arr.shape()
+  arr.size()
+
+-----
+
+You can get the dimensions of a NumPy array any time using ``ndarray.shape``. NumPy will return the dimensions of the array as a tuple.
+
+For example, if you create this array:
+
+::
+
+  arr = np.array([[1 , 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+
+You can use ``arr.shape`` to find the shape of your array.
+
+**Input:**
+
+::
+
+  arr.shape
+
+**Output:**
+
+::
+
+  (3, 4)
+
+This output tells you that your array has three rows and four columns.
+
+You can find just the number of rows by specifying ``[0]``:
+
+**Input:**
+
+::
+
+  num_of_rows = arr.shape[0]
+ 
+  print('Number of Rows : ', num_of_rows)
+
+**Output:**
+
+::
+
+  Number of Rows :  3
+
+Or just the number of columns by specifying ``[1]``:
+
+**Input:**
+
+::
+
+  num_of_columns = arr.shape[1]
+ 
+  print('Number of Columns : ', num_of_columns) 
+
+**Output:**
+
+::
+  
+  Number of Columns :  4
+
+It's also easy to find the total number of elements in your array:
+
+**Input:**
+
+::
+
+  print(arr.shape[0] * arr.shape[1])
+
+**Output:**
+
+::
+
+  12
+
+You can use ``arr.shape()`` with a 1D array as well. If you create this array:
+
+**Input:**
+
+::
+
+  arr = arr.array([1, 2, 3, 4, 5, 6, 7, 8])
+
+You can print the shape and the length of the array.
+
+::
+
+  print('Shape of 1D array: ', arr.shape)
+  print('Length of 1D array: ', arr.shape[0])
+
+**Output:**
+
+::
+
+  Shape of 1D array:  (8,)
+  Length of 1D array:  8
+
+
+You can get the dimensions of an array using ``arr.size()``.
+
+**Input:**
+
+::
+
+  # get number of rows in array
+  num_of_rows2 = np.size(arr, 0)
+ 
+  # get number of columns in 2D numpy array
+  num_of_columns2 = arr.size(arr, 1)
+ 
+  print('Number of Rows : ', num_of_rows2)
+  print('Number of Columns : ', num_of_columns2)
+
+**Output:**
+
+::
+
+  Number of Rows :  3
+  Number of Columns: 4
+
+You can print the total number of elements as well:
+
+**Input:**
+
+::
+  
+  print('Total number of elements in  array : ', arr.size(arr))
+
+**Output:**
+
+::
+
+  Total number of elements in  array :  12
+
+This also works for 3D arrays:
+
+**Input:**
+
+::
+
+  arr3D = np.array([ [[1, 1, 1, 1], [2, 2, 2, 2], [3, 3, 3, 3]],
+                 [[4, 4, 4, 4], [5, 5, 5, 5], [6, 6, 6, 6]] ])
+
+You can easily print the size of the axis:
+
+**Input:**
+
+::
+
+  print('Axis 0 size : ', arr.size(arr3D, 0))
+  print('Axis 1 size : ', arr.size(arr3D, 1))
+  print('Axis 2 size : ', arr.size(arr3D, 2))
```

Comment:
same, should be `arr3D.shape[...]`.

Reply:
also, could you name the array `arr_3d`?

Reply:
Done!